pkg/project/apiserver/admission/nodeenv/admission.go

```diff
@@ -67,12 +69,22 @@ func (p *podNodeEnvironment) Admit(a admission.Attributes) (err error) {
 		return apierrors.NewForbidden(resource, name, fmt.Errorf("pod node label selector conflicts with its project node label selector"))
 	}
 
-	// modify pod node selector = project node selector + current pod node selector
-	pod.Spec.NodeSelector = labelselector.Merge(projectNodeSelector, pod.Spec.NodeSelector)
+	if mutationAllowed {
+		// modify pod node selector = project node selector + current pod node selector
+		pod.Spec.NodeSelector = labelselector.Merge(projectNodeSelector, pod.Spec.NodeSelector)
+	}
```

Comment:
needs an else to fail if the merged selector is not identical to the pod selector, right?

Reply:
hah, you were 1 min faster :)

Reply:
done